arraykit.c

```diff
@@ -18,6 +21,11 @@
                             Py_TYPE(O)->tp_name);                             \
     }
 
+# define AK_NOT_IMPLEMENTED                                                   \
+    PyErr_SetNone(PyExc_NotImplementedError);                                 \
+    return NULL;                                                              \
```

Comment:
Time for another C-ism: macros which are meant to be used as a single statement *should be written as a single statement*.

It's not required, but it is generally encouraged. For example, in projects that don't require braces for single-statement conditionals, this...

```c
if (condition) AK_NOT_IMPLEMENTED;
```

would actually expand to:

```c
if (condition) PyErr_SetNone(PyExc_NotImplementedError);
return NULL;
```

...which is subtly different (but entirely wrong)!

The easiest way to do this is `do { ... } while (0)`:

```suggestion
    do {                                                                      \
        PyErr_SetNone(PyExc_NotImplementedError);                             \
        return NULL;                                                          \
    } while (0)
```

(It might seem like `if (1) { ... }` would also work here, but expanding that can mess up `if`-`elif-`else` chains in even nastier ways!)

Also, no slash on the last line!

Reply:
Thanks for the explanation!